tensorflow/compiler/mlir/lite/ir/tfl_ops.cc

```diff
@@ -955,7 +956,7 @@ static LogicalResult Verify(ScatterNdOp op) {
     // Checks whether the last `(shape_type.getDimSize(0) - outermost_dim)`
     // dimensions of `updates` and `shape` are equal.
     for (auto shape_it : llvm::enumerate(shape_value)) {
-      auto i = shape_it.index();
+      long int i = shape_it.index();
```

Comment:
Is this really needed? I would expect the compiler to determine the right type for `auto`, since `i` has only a small number of uses throughout its timeline.

Let's exclude from the PR and revisit at the end.

Reply:
This was done to resolve [-Wsign-compare] that occurs in `if (i >= outermost_dim) {`

Reply:
They're both declared as auto, however these are the observed types:
```
tensorflow/compiler/mlir/lite/ir/tfl_ops.cc:783:13: warning: comparison of integer expressions of different signedness: **'long unsigned int'** and **'long int'** [-Wsign-compare]
  783 |       if (i >= outermost_dim) {
```

`i` is the unsigned type.